File: mobatech-flutter/lib/features/appointment/data/models/polyclinic.dart
class PolyclinicSchedule {
  final int id;
  final int polyclinicId;
  final String dayOfWeek;
  final String startTime;
  final String endTime;
  final bool isAvailable;

  PolyclinicSchedule({
    required this.id,
    required this.polyclinicId,
    required this.dayOfWeek,
    required this.startTime,
    required this.endTime,
    required this.isAvailable,
  });

  factory PolyclinicSchedule.fromJson(Map<String, dynamic> json) {
    return PolyclinicSchedule(
      id: json['ID'] ?? 0,
      polyclinicId: json['polyclinic_id'] ?? 0,
      dayOfWeek: json['day_of_week'] ?? '',
      startTime: json['start_time'] ?? '',
      endTime: json['end_time'] ?? '',
      isAvailable: json['is_available'] ?? false,
    );
  }
}

class Polyclinic {
  final int id;
  final String name;
  final String description;
  final String imageUrl;
  final bool isActive;
  final List<PolyclinicSchedule> schedules;

  Polyclinic({
    required this.id,
    required this.name,
    required this.description,
    required this.imageUrl,
    required this.isActive,
    required this.schedules,
  });

  factory Polyclinic.fromJson(Map<String, dynamic> json) {
    var list = json['schedules'] as List? ?? [];
    List<PolyclinicSchedule> schedulesList = list
        .map((i) => PolyclinicSchedule.fromJson(i))
        .toList();

    return Polyclinic(
      id: json['ID'] ?? 0,
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      imageUrl: json['image_url'] ?? '',
      isActive: json['is_active'] ?? false,
      schedules: schedulesList,
    );
  }
}

File: mobatech-backend/models/polyclinic.go
package models

import "gorm.io/gorm"

type Polyclinic struct {
	gorm.Model
	Name        string               `json:"name"`
	Description string               `json:"description"`
	ImageURL    string               `json:"image_url"`
	IsActive    bool                 `json:"is_active" gorm:"default:true"`
	Schedules   []PolyclinicSchedule `json:"schedules,omitempty" gorm:"foreignKey:PolyclinicID"`
}

type PolyclinicSchedule struct {
	gorm.Model
	PolyclinicID uint   `json:"polyclinic_id"`
	DayOfWeek    string `json:"day_of_week"` // e.g. "Senin", "Selasa"
	StartTime    string `json:"start_time"`  // e.g. "08:00"
	EndTime      string `json:"end_time"`    // e.g. "16:00"
	IsAvailable  bool   `json:"is_available" gorm:"default:true"`
}
